File: frontend/lib/models/house.dart
class House {
  final String id;
  final String name;
  final String inviteCode;
  final String createdBy;
  final List<String> memberIds;
  final Map<String, String> memberNames;

  House({
    required this.id,
    required this.name,
    required this.inviteCode,
    required this.createdBy,
    required this.memberIds,
    required this.memberNames,
  });

  factory House.fromJson(Map<String, dynamic> json) {
    return House(
      id: json['id'],
      name: json['name'],
      inviteCode: json['inviteCode'],
      createdBy: json['createdBy'],
      memberIds: List<String>.from(json['memberIds'] ?? []),
      memberNames: Map<String, String>.from(json['memberNames'] ?? {}),
    );
  }
}

File: backend/src/main/java/com/casacontrole/model/House.java
package com.casacontrole.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class House {
    private String id;
    private String name;
    private String inviteCode;       // código de 6 chars para entrar na casa
    private String createdBy;        // uid do criador
    private List<String> memberIds;  // lista de uids
    private Map<String, String> memberNames; // uid -> nome
    private Instant createdAt;
}
